ReSharper.FSharp/src/FSharp.Psi.Daemon/src/QuickDoc/FSharpQuickDocProvider.fs

```diff
@@ -24,9 +24,16 @@ module FSharpQuickDoc =
 
         // todo: fix getting qualifiers
         let tokenNames = [token.Name]
+        let treeEndOffset =
+            if (isNull token.Parent) then token.GetTreeEndOffset() else
+            if (isNull token.Parent.Parent) then token.GetTreeEndOffset() else
+            let barTreeNode = token.Parent.Parent.Children() |> Seq.tryFindBack(isBar)
+            match barTreeNode with
+            | Some n -> n.GetTreeEndOffset()
```

Comment:
We're lucky that there are currently no other rules in the grammar that would match it, but if we change a structure of anything that contains `|` then it may suddenly match this check. Consider:
```fsharp
type E =
    | A = 1
```
```fsharp
type U =
    | U of i: int
```
```fsharp
match () with
| x -> ()
```
```fsharp
match () with
| a | b -> ()
```

I can imagine that we could want to remove the field list node in the second case in the future (i.e. move the field nodes to the case node), and that would suddenly break the tooltips for `i` because of this check.